user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/GQIPage.md

```diff
@@ -17,3 +17,36 @@ Represents a single page of rows. Every query result consists of one or more suc
 | -------- | ---- | -------- | ----------- |
 | Rows | [GQIRow](xref:GQI_GQIRow)[] | Yes | Rows of the page. |
 | HasNextPage | boolean  | No | `True` if GQI can fetch additional pages, `False` if the current page is the last page. |
+
+> [!TIP]
+> To enhance performance and efficiency, consider restricting the data returned by using the `HasNextPage` boolean of a `GQIPage`. Often, not all data from an ad hoc data source is essential for the visualization. Utilizing paged data retrieval makes sure the implementation only retrieves the necessary pages until GQI accumulates sufficient data. This approach offers several benefits:
+> - Speed: Fetching all data immediately in your implementation can be inefficient. By retrieving only the required pages, you optimize processing speed.
+> - Memory Efficiency: Limiting the data retrieved helps conserve memory resources. Unnecessary data won't burden the system, allowing it to operate more efficiently.
+
+### Examples
+
+#### Paged data retrieval
+
+The following example illustrates how the `HasNextPage` boolean can be manipulated to indicate that there are additional pages. If `HasNextPage` is set to `False`, GQI might call `GetNextPage` once more, until it has enough data.
+
+```csharp
+private int pageCounter = 0;
+private int pageSize = 100;
+
+public GQIPage GetNextPage(GetNextPageInputArgs args)
+{
+    var rows = FetchData(pageCounter, pageSize);
+    var hasNextPage = rows.Length == pageSize;
```

Comment:
Maybe this needs a comment to explain that we assume there is more data if we get a full page.